### PR TITLE
[DOCS] Explain high s value

### PR DESCRIPTION
**Pull request by ChrisChinchilla**
### Description

Someone helpfully supplied a resource for the high-s value (https://github.com/ethereum/solidity/pull/6200#discussion_r269549494), so this PR adds that detail.

### Checklist
- [x] Code compiles correctly
- [x] All tests are passing
- [x] New tests have been created which fail without the change (if possible)
- [x] README / documentation was extended, if necessary
- [x] Changelog entry (if change is visible to the user)
- [x] Used meaningful commit messages

### CODE REVIEW

File: docs/units-and-global-variables.rst [hunk 1]
```diff
@@ -186,6 +186,11 @@ Mathematical and Cryptographic Functions
 
     For further details, read `example usage <https://ethereum.stackexchange.com/q/1777/222>`_.
 
+.. note::
+
+    Since `EIP-2 <http://eips.ethereum.org/EIPS/eip-2#specification>`_, all transaction signatures with an ``s`` value
```
Comment (chriseth) [inline docs/units-and-global-variables.rst]: I think this should be merged with the following warning. Please also clarify that EIP2 only applies to transaction signatures and not to the function/precompile ecrecover.

Comment (ChrisChinchilla) [reply docs/units-and-global-variables.rst]: Good idea, I think the last commit covers this.

File: docs/units-and-global-variables.rst [hunk 2]
```diff
@@ -189,7 +189,10 @@ Mathematical and Cryptographic Functions
 .. warning::
 
     If you use ``ecrecover``, be aware that a valid signature can be turned into a different valid signature without requiring
-    knowledge of the corresponding private key. This is usually not a problem unless you require signatures to be unique or
+    knowledge of the corresponding private key. Since `EIP-2 <http://eips.ethereum.org/EIPS/eip-2#specification>`_, all transaction signatures with an ``s`` value
```
Comment (chriseth) [inline docs/units-and-global-variables.rst]:
```suggestion
    knowledge of the corresponding private key. For transactions, `EIP-2 <http://eips.ethereum.org/EIPS/eip-2#specification>`_ fixed this problem by requiring the ``s`` value to be greater than ``secp256k1n/2``, but this requirement is not enforced by the ``ecrecover`` precompile to stay compatible with other uses.
```

File: docs/units-and-global-variables.rst (hunk 1 above)
Comment (ChrisChinchilla) [reply docs/units-and-global-variables.rst]: Looks good @chriseth

File: docs/units-and-global-variables.rst [hunk 3]
```diff
@@ -189,7 +189,10 @@ Mathematical and Cryptographic Functions
 .. warning::
 
     If you use ``ecrecover``, be aware that a valid signature can be turned into a different valid signature without requiring
-    knowledge of the corresponding private key. This is usually not a problem unless you require signatures to be unique or
+    knowledge of the corresponding private key. For transactions, `EIP-2 <http://eips.ethereum.org/EIPS/eip-2#specification>`_ fixed this problem by requiring the ``s`` value to be greater than ``secp256k1n/2``, but this requirement is not enforced by the ``ecrecover`` precompile to stay compatible with other uses.
+    greater than ``secp256k1n/2`` are considered invalid.
```
Comment (leonardoalt) [inline docs/units-and-global-variables.rst]: This line looks off

Comment (ChrisChinchilla) [reply docs/units-and-global-variables.rst]: Tidied a little @leonardoalt

**Review by leonardoalt — APPROVED** [review]

**Pull request merged**